scripts/lib/CIME/XML/pes.py

```diff
@@ -96,7 +96,7 @@ def _find_matches(self, grid_nodes, grid, compset, machine, pesize_opts, overrid
                             pesize_match = self.get(pes_node, "pesize")
                             compset_match = self.get(pes_node, "compset")
                             if (pesize_match == "any" or (pesize_opts is not None and \
-                                                          re.search(pesize_match, pesize_opts))) and \
+                                                          pesize_match == pesize_opts)) and \
```

Comment:
This is a meaningful change.

Reply:
@amametjanov explained this to me:

re.search(pattern, string) returns true if the regular expression pattern is a substring anywhere in the string. If there are 2 or more PE layouts with pesize=x and pesize=y, where x is a substring of y or y is a substring of x, then these PE layouts have the same match score and the resulting error is

    Errors were:
        mach_choice theta mach_match theta
        grid_choice a%ne120np4 grid_match a%ne120np4
        compset_choice .*CAM5.+CLM45.+MPASCICE.+MPASO.+MOSART.* compset_match .*CAM5.+CLM45.+MPASCICE.+MPASO.+MOSART.*
        pesize_choice L pesize_match XL  
        points = 13
        ERROR: More than one PE layout matches given PE specs
I think pecount/pesize naming conventions are

S - small
M/any/default - medium
L - large
XL - extra-large
XS - extra-small
T - threaded
So this PR tries to follow that convention with pesize=MT for medium-threaded and pesize=XLT for extra-large-threaded PE layouts. This requires change from regular expressions to exact equality, because L is a substring of XL.

Reply:
I have been arguing that we had too many possibilities here and should just reduce it to S,M,L but you are going in the opposite direction.   As I write this I realize that it's not relevant to this change.

Reply:
fwiw, I've been wanting more options here as well.  For one, I think having more sizes is always better (T, XL, X1, X2?)  And, for example, we are thinking of having different PE layouts based on the amount of high frequency output.  Say T for tiny and THF for tiny with high-frequency output.  OR, maybe not call out HF output, but something like favoring using less memory, ie fewer MPI's, more threads.  TDEBUG?  I don't know.